### PR TITLE
Make ebooks unique. 

### LINKED ISSUE #1: Change 11ty so that it does not combine ebooks
The JS bits are currently broken when sorting ebook pages when more than one ebook is created from the template. They get integrated into one group when they need to be separate groups. 

Goal: make sure one could copy the sample-ebook inside `/src/ebooks` and just add new content and images. Rinse, repeat.

Line 9 of .eleventy.js is where one of the problems is. It grabs all the collections and sorts them as one giant thing. Needs to be updated to do each collection individually so the ebooks don’t get mixed.

### PR DESCRIPTION
**Pull request by sirkitree**
Resolves #1

This was a beast.

In short terms the relevant pieces are:
* a different way to create a multidimensional array of books and their chapters
* some custom pagination because eleventy pagination out of the box doesn't support multidimensional arrays
* a more standard approach to using templates rather than conditionally including them based on pagination
* blood sweat and tears

Note that this is based on my first pull request, so you'll want to merge that first.

### CODE REVIEW

**Comment by sirkitree** [comment]
or perhaps this has all those changes, so maybe just merge this one? i don't git much these days so i perhaps messed up branching

File: package.json [hunk 1]
```diff
@@ -19,5 +19,8 @@
     "@11ty/eleventy-img": "^4.0.2",
     "npm-run-all": "^4.1.5",
     "sass": "^1.75.0"
+  },
+  "dependencies": {
+    "sharp": "^0.33.4"
```
Comment (huntermacd) [inline package.json:24]: Is this being used? I think `eleventy-img` already provides this utility.

File: src/_includes/layout.liquid [hunk 2]
```diff
@@ -11,20 +11,25 @@
...
         <div>{{ chapter.data.title }}</div>
       </div>
-      {% if pagination.pageNumber != 0 %}
+      
+    {% assign currentBookSlug = page.filePathStem | split: "/" | slice: 2, 1 | join: "" %}  
+    {% for bookWithChapters in collections.booksWithChapters %}
+      {% if bookWithChapters.book.slug == currentBookSlug %}
         <div class="page-number">
-          <div>{% if pagination.pageNumber == 1 %}TOC{% else %}{{ pagination.pageNumber | minus: 1 }}{% endif %}</div>
+          <div><a href="/ebooks/{{ bookWithChapters.book.slug }}/table-of-contents">TOC</a></div>
```
Comment (huntermacd) [inline src/_includes/layout.liquid:28]: We want this to be blank on the cover, say "TOC" on the TOC, and switch to page number for the chapters.

File: src/ebooks/.DS_Store
Comment (huntermacd) [inline src/ebooks/.DS_Store:1]: This file can be removed. Maybe we can add `.DS_Store` to the `.gitignore`.

File: src/ebooks/my-sample-ebook/chapters/chapter-2.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
 title: "Chapter 2"
-tags: chapter
+layout: chapter
```
Comment (huntermacd) [inline src/ebooks/my-sample-ebook/chapters/chapter-2.md:3]: Let's use the `.liquid` extension when specifying the layout everywhere or remove them everywhere for consistency.

**Review by huntermacd — CHANGES_REQUESTED** [review]
I noticed for the `my-sample-ebook` has only two chapters but there are three items in the TOC and you can reach chapter three which is just chapter again but without any styling.

Added some questions and suggestions. Otherwise, looks great! I'm able to add another ebook to the directory and it shows up in on the site with proper page ordering. Great work!

File: package.json (hunk 1 above)
Comment (sirkitree) [reply package.json:24]: I'm honestly not sure, but originally when I first cloned the repo and tried to `npm install` I kept getting an error about this and couldn't proceed until I installed it. 🤷

File: src/_includes/layout.liquid (hunk 2 above)
Comment (sirkitree) [reply src/_includes/layout.liquid:28]: Ah, ok. I thought it was meant to be a quick way to get back to the TOC.

File: src/ebooks/.DS_Store
Comment (sirkitree) [reply src/ebooks/.DS_Store:1]: ah, yes I think I removed this in my subsequent PR

**Comment by sirkitree** [comment]
> I noticed for the my-sample-ebook has only two chapters but there are three items in the TOC and you can reach chapter three which is just chapter again but without any styling.
Monosnap Table of Contents 2024-06-11 06-30-22


I'm not seeing this...

**Comment by sirkitree** [comment]
addressed all suggestions above - please re-review.

File: package.json (hunk 1 above)
Comment (huntermacd) [reply package.json:24]: Hm. Fair enough. :)

File: src/ebooks/.DS_Store
Comment (huntermacd) [inline src/ebooks/.DS_Store:1]: Still here. :D

**Review by huntermacd — COMMENTED** [review]
Looks great! Still seeing that .DS_Store file. Remove that and we can :shipit:.

**Comment by huntermacd** [comment]
If you delete the _site directory and rebuild, do you still not see the erroneous third chapter for the sample ebook?

File: src/ebooks/.DS_Store
Comment (sirkitree) [reply src/ebooks/.DS_Store:1]: hrm, wtf... I don't have one locally... and if I do a fresh checkout it doesn't create one. I'm not sure how to remove this...

**Comment by sirkitree** [comment]
Correct, if I delete `_site`, `npm run build`, and then close down the server and restart it with `npm run start` I get this: 
image

**Comment by sirkitree** [comment]
merging so I can keep building

**Pull request merged**